toolbox/timecomparer/main.go

```
@@ -0,0 +1,537 @@
+	var timeArray []float64
+	commandsBreakDown := map[float64]commandAndTime{}
+	for command, rCombination := range commandToTime {
+		time := rCombination.totalTime
+		rTimes := rCombination.runTimes
+		averageTime := time / float64(len(rTimes))
+		cAndTime := commandAndTime{
+			command:    command,
+			runStrings: []string{},
+		}
+		for _, r := range rTimes {
+			cAndTime.runStrings = append(cAndTime.runStrings, r.toStringSlice()...)
+		}
+		timeArray = append(timeArray, averageTime)
+		commandsBreakDown[averageTime] = cAndTime
+	}
+	return timeArray, commandsBreakDown
+}
+
+// Convert minutes and sesconds ("0m1.061s") to float object.
+func convertStringToTime(timeString string) ([]float64, error) {
+	var timeSlice []float64
+	if strings.Contains(timeString, "m") {
+		timeSplit := strings.Split(timeString, "m")
+		minString := timeSplit[0]
+		fmt.Println("get minutes")
+		fmt.Println(minString)
+		minInt, err := strconv.Atoi(minString)
+		if err != nil {
+			fmt.Println("error parsing float minute")
+			timeSlice = append(timeSlice, -1.0)
+		} else {
+			minFloat := float64(minInt)
+			timeSlice = append(timeSlice, minFloat)
+		}
+		if len(timeSplit) > 1 {
+			secString := timeSplit[len(timeSplit)-1]
+			secValueString := strings.Split(secString, "s")[0]
+			fmt.Println("get seconds")
+			fmt.Println(secValueString)
+			secFloat, err := strconv.ParseFloat(secValueString, 64)
+			if err != nil {
+				return timeSlice, fmt.Errorf("error parsing float second")
+			}
+			timeSlice = append(timeSlice, secFloat)
+		}
+	}
+	return timeSlice, nil
+}
+
+// Extract time from runtime string with the format of "real	0m1.061s"
+func extractTime(runTimeString string) runPair {
+	timeSplit := strings.Fields(runTimeString)
+	length := len(timeSplit)
+	var ind int
+	for ind = length - 1; ind > 0; ind-- {
+		if strings.Compare(timeSplit[ind], "") != 0 {
+			break
+		}
+	}
+	timePart := timeSplit[ind]
+	timeSlice, _ := convertStringToTime(timePart)
```

don't ignore the error, please